### DISCUSSION TITLE
How to bulk resolve ANS name

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#126] [Questions] [answered]
Asked by `cryptisen` on discord dev-discussion
--- 
hey team, quick question for aptosnames restapi where is all the documentation? checking for a bulk primary domain resolver 😄
I want to resolve a lot of address to primary names and cache those in backend but cannot find it (a bulk resolver), no way of doing it 1 by 1 right? 😄

**Comment by 0x-j** [comment, marked as answer]
I don't think we have a batch call API, so you need to create your own contract with a view function that batch calls the resolve name function in ANS.  

[Here's an example contract I wrote](https://github.com/0xaptosj/playground/blob/main/sources/ans_playground.move#L11-L28).
[Here's how I call my contract using ts sdk](https://github.com/0xaptosj/playground/blob/main/src/utils/aptos.ts#L46-L57).

**Reply by Cryptisen** [reply]
Thanks! I will give a try this weekend
